packages/parser-package-json/src/parser.ts

```diff
@@ -0,0 +1,77 @@
+import * as path from 'path';
+
+import { FetchEnd, IJSONResult, Parser, SchemaValidationResult } from 'hint/dist/src/lib/types';
+import { Engine } from 'hint';
+import loadJSONFile from 'hint/dist/src/lib/utils/fs/load-json-file';
+import { parseJSON } from 'hint/dist/src/lib/utils/json-parser';
+import { validate } from 'hint/dist/src/lib/utils/schema-validator';
+
+import { PackageJsonEvents } from './types';
+
+export * from './types';
+
+export default class PackageJsonParser extends Parser<PackageJsonEvents> {
+    private schema: any;
+
+    public constructor(engine: Engine<PackageJsonEvents>) {
+        super(engine, 'package-json');
+        this.schema = loadJSONFile(path.join(__dirname, 'schema.json'));
```

Comment:
Where did we get the schema definition from? A comment would be useful here so we know where to look when we need to update it.

Reply:
Also, does this schema allow additional custom properties at the root? I know a number of packages have taken to extending package.json (e.g. [browserslist](https://github.com/browserslist/browserslist#browserslist-)) and I don't think we want to fail validation in that case.

Reply:
AFAIK JsonSchema allow you to validate properties defined in the schema, if you add some extra properties not considered in the schema, these properties won't be validated against the schema, so the json should still be valid..

Reply:
@antross There is a test that checks for this - line 132 in tests/tests.ts `If 'package.json' contains a non standard property, it should still emit the event parse::end::package-json`

Reply:
I've updated the test using the browserslist example...

Reply:
Great, thanks for clarifying!